Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-
-	//Room4
-	rooms[3].setNorth(&rooms[0]);
-	rooms[3].setItemsPresent(0 , items[1]);
-
-
-	//Room5
-	rooms[4].setNorth(&rooms[1]);
-	rooms[4].setEast(&rooms[5]);
-
-	rooms[4].setMonsterPresent( monsters[0] );
-
-
-	//Room6
-	rooms[5].setWest(&rooms[4]);
-
-	rooms[5].setMonsterPresent( monsters[1] );
-
-
-	//Room7
-	rooms[6].setEast(&rooms[7]);
-	rooms[6].setItemsPresent(0 , items[4]);
-
-
-	//Room8
-	rooms[7].setWest(&rooms[6]);
-	rooms[7].setNorth(&rooms[4]);
-
-	rooms[7].setItemsPresent(0 , items[2]);
-
-
-	//Room9
-	rooms[8].setNorth(&rooms[5]);
-	rooms[8].setPrincessPresent(princess);
-}
-
-
-
-void Castle::LinkRoom5and8()
-{
-	rooms[4].setSouth(&rooms[7]);
-	rooms[7].setNorth(&rooms[4]);
-}
-
-
-
-void Castle::LinkRoom6and9()
-{
-	rooms[5].setSouth(&rooms[8]);
-	rooms[8].setNorth(&rooms[5]);
-}
-
-
-
-void Castle::HiddenRoomsUnlocker(char const* monsterName , Monster** monsters)
-{
-	if (strcmp(monsterName , (monsters[0]->getMonsterName())) == 0)
-	{
-		LinkRoom5and8();
-	}
-
-	else if (strcmp(monsterName , (monsters[1]->getMonsterName())) == 0)
-	{
-		LinkRoom6and9();
-	}
-}
-
-
-
-void Castle::setDescriptionOfRooms()
-{
-	ifstream InputStream("Rooms.txt");
-	if (InputStream.is_open())
-	{
-		char buffer[300];
-
-		for (int i = 0; i < 9 && !InputStream.eof(); ++i)
-		{
-			InputStream.getline(buffer,300);
-			rooms[i].setDesription(buffer);
-		}
-	}
-	else
-		cout << "Could not open file";
-	InputStream.close();
-}
-
-
-
-void Castle::setNumbersofRooms()
-{
-	for (int i = 0; i < 9; ++i)
-	{
-		rooms[i].setRoomNumber(i+1);
-	}
-}
-
-
-
-Castle::~Castle()
-{
-
-}
-
-
-
-//-------------------Player---------------------
-
-
-
-Player::Player(char const* player_Name , Room* room1 )
-{
-	name = new char [strlen(player_Name) + 1];
-	strcpy_s (name,strlen(player_Name) + 1, player_Name);
-
-	current_Room = room1;
-	princess_Pointer = nullptr;
-	is_Alive = true;
-	cash = 0;
-
-	for (int i = 0; i < 10; ++i)
-	{
-		bag[i] = nullptr;
-	}
-}
-
-
-
-Player::Player()
-{
-
-}
-
-
-
-Player::~Player()
-{
-	if (name)
-	{
-		delete[] name;
-	}
-}
-
-
-
-bool Player::getCurrentState()
-{
-	return is_Alive;
-}
-
-
-
-void Player::setCurrentState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-Princess* Player::getPrincess_Pointer()
-{
-	return princess_Pointer;
-}
-
-
-
-char* Player::getPlayerName()
-{
-	return name;
-}
-
-
-
-void Player::CashUpdater()
-{
-	cash = 0;
-
-	for (int i = 0; i < 10; ++i)
-	{
-		if(bag[i])
-		{
-			cash = cash + bag[i]->getWorth();
-		}
-	}
-}
-
-
-
-int Player::getCash()
-{
-	return cash;
-}
-
-
-
-bool Player::isBagEmpty()
-{
-	for (int i = 0; i < 10; ++i)
-	{
-		if(bag[i] != nullptr)
-			return false;
-	}
-
-	return true;
-}
-
-
-
-bool Player::isBagFull()
-{
-	bool bag_Full = true;
-
-	for (int i = 0; i < 10 && bag_Full; ++i)
-	{
-		if(bag[i] == nullptr)
-			bag_Full = false;
-	}
-
-	return bag_Full;
-}
-
-
-
-bool Player::Move(char const* direction , bool& exit_Castle)
-{
-	bool moveSuccessful = false;
-	int i = 0;
```

Comment:
Why set i to 0, then immediately set it to another value.  Combine this and line 712 into a single statement.